### ISSUE TITLE
XingClient not working on ios9

### ISSUE THREAD

**Issue by pranav18** [xing/XNGAPIClient#77]
Hello,
I am using same client as I have used in iOS 8.4 but client is not working in iOS 9.
Same code is working in iOS 8.4 but it is not working in iOS 9 I don't know but it is weird.
As from iOS 9 we have to set security key in info.plist which I am setting but still client is still not working.
I am getting below error : 
Error Domain=com.alamofire.error.serialization.response Code=-1011 "Request failed: unauthorized (401)" UserInfo={com.alamofire.serialization.response.error.response=<NSHTTPURLResponse: 0x7fbf8be25790> { URL: https://api.xing.com/v1/request_token } { status code: 401, headers {
    "Cache-Control" = "no-cache";
    Connection = "Keep-Alive";
    "Content-Encoding" = gzip;
    "Content-Length" = 93;
    "Content-Type" = "application/json; charset=utf-8";
    Date = "Tue, 01 Sep 2015 06:29:16 GMT";
    "Keep-Alive" = "timeout=5, max=56";
    Server = Apache;
    "Set-Cookie" = "c_=baca882ff97bd2cf6a18b395933613a0; domain=.xing.com; path=/; expires=Mon, 27 Aug 2035 06:29:16 -0000; secure; HttpOnly";
    Status = "401 Unauthorized";
    Vary = "Accept-Encoding";
    "X-Content-Type-Options" = nosniff;
    "X-Frame-Options" = SAMEORIGIN;
    "X-Logjam-Request-Action" = "WebService::OauthController#request_token";
    "X-Logjam-Request-Id" = "xws-production-85b5a502132e496daedb58968863026a";
    "X-Powered-By" = "Phusion Passenger 4.0.59";
    "X-Request-Id" = "e82a4a5e-5876-4e1d-9017-85dbbbd1a0f4";
    "X-Runtime" = "0.012464";
    "X-XSS-Protection" = "1; mode=block";
} }, NSErrorFailingURLKey=https://api.xing.com/v1/request_token, com.alamofire.serialization.response.error.data=<7b226d65 73736167 65223a22 496e7661 6c696420 4f417574 68207369 676e6174 75726522 2c226572 726f725f 6e616d65 223a2249 4e56414c 49445f4f 41555448 5f534947 4e415455 5245227d>, NSLocalizedDescription=Request failed: unauthorized (401)}

I am have below code in my info.plist:
<key>NSAppTransportSecurity</key>
	<dict>
		<key>NSAllowsArbitraryLoads</key>
		<true/>
	</dict>

**Comment by PatrickV2** [comment]
Seems that I have the same issue. I tried to find a solution but I can not find one.

**Comment by matthias-maennich** [comment]
Can you reproduce it with our Example project? I tried and it worked fine after setting the App Transport Security rule to NSAllowsArbitraryLoads.

**Comment by PatrickV2** [comment]
Yes, I can reproduce it in the example project even after applying NSAllowArbitraryLoads. I am using XCode 7 beta 6.
I get the following NSError:
![screen shot 2015-09-06 at 12 49 48](https://cloud.githubusercontent.com/assets/10108665/9704011/5bb8829c-5496-11e5-86d7-d3b3f2cef69c.png)

**Comment by toco** [comment]
Why should ```NSAllowsArbitraryLoads``` be necessary at all? The XING API always uses TLS/https, right?
What does your the "Invalid requests" section in the dashboard for your application on [dev.xing.com](https://dev.xing.com/applications/dashboard) say?
It looks more like an issue with the OAuth signature, token or key…

**Comment by PatrickV2** [comment]
From what I understood you still have to specify in iOS9 the sites that you are allowed to access (NSAPPTransportSecurity). With NSAllowsArbitraryLoads you just specify that everything is allowed. I will need to rework that as soon as I get this up and running again.
My "invalid requests" dashboard for my application does not contain any error related to this issue (there are only some old ones when I was building in initial support for Xing).

**Comment by toco** [comment]
In tests, my code worked in the iPhone Simulator with Xcode 7 beta 6 with following ATS exceptions:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSExceptionDomains</key>
	<dict>
		<key>xing.com</key>
		<dict>
			<key>NSIncludesSubdomains</key>
			<true/>
			<key>NSTemporaryExceptionRequiresForwardSecrecy</key>
			<false/>
		</dict>
		<key>xingassets.com</key>
		<dict>
			<key>NSIncludesSubdomains</key>
			<true/>
			<key>NSTemporaryExceptionRequiresForwardSecrecy</key>
			<false/>
		</dict>
	</dict>
</dict>
</plist>
```

**Comment by PatrickV2** [comment]
I added the keys to the plist but it has the same result on my project and the example project. Both in the simulator and on my iPhone 6.

**Comment by pranav18** [comment]
Hello guys,
Finally I found the resolution of issue that we are facing in XingClient.
Issue is from AFNetwork.
AFNetwork that we are using is not supporting properly in XingClient.
So please download AFNetwork form below URL (If you are using pod so you can replace it pod AFNetwork folder).

https://www.dropbox.com/s/qh4vrbd4qkgrgdj/AFNetworking.zip?dl=0

just add one flag in plist NSAllowsArbitraryLoads than it will work.
If still it is not working I can share my code also.

**Comment by PatrickV2** [comment]
I will test this later but which version of AFNetworking is this?
Is this a standard release of AFNetworking or did you fix something in AFNetworking?
If so, what did you change?

**Comment by PatrickV2** [comment]
Going back to AFNetworking version 2.5.4 also resolved the issue for me. It seems like that version 2.6.0 of AFNetworking is not compatible with the XingClient.

**Comment by pranav18** [comment]
What the hell is going on with Xing API.
Again it stop working, Is any body is doing black magic, its weird.
I don't know what should i do. I am blank.
Please help me. My app is going for Release.

**Comment by PatrickV2** [comment]
Did you try to get AFNetworking 2.5.4 instead of 2.6? That fixed it for me...

This message was sent from my phone so it is probably short and to the point.

**Comment by pranav18** [comment]
Yes I have tried it was working previously but its not working now.
Also I am not getting any error for now.
Nothing is happing in this block :

    [[XNGAPIClient sharedClient] loginOAuthWithSuccess:^{
        // handle success
        
    }
                          failure:^(NSError *error) {
                              // handle failure
                              NSLog(@"%@",error);
                          }];


Its working great in iOS 8.4 but not getting any response in iOS 9.

**Comment by Ruenzuo** [comment]
@pranav18 what are you using to handle authentication? Safari or UIWebView?

**Comment by pranav18** [comment]
Safari

**Comment by PatrickV2** [comment]
Using webview and for me it is still working with version 2.5.4.

This message was sent from my phone so it is probably short and to the point.

**Comment by Ruenzuo** [comment]
Just checked the example project and everything is working fine with `NSAllowsArbitraryLoads` and Safari authentication. 
With `UIWebView` I had to implement a delegate method to handle the request with the scheme (xingapp~consumerkey~) and call `-[XNGAPIClient handleOpenURL:]`.
We'll update our example project to include iOS 9 instructions.

I would like to provide additional help, but I'm unable to reproduce this error. If you can reproduce this error on the example project, please commit your changes into a branch and we'll check it. Else, feel free to send me an xcodeproj with the reproducible error to my email address.
